### DISCUSSION TITLE
How to record from more cameras than USB ports?

### DISCUSSION THREAD

**Discussion by mjhmilla** [freemocap/freemocap#259] [Q&A] [answered]
I would like to record from as many as 6 usb webcams from a laptop using freemocap. As many have found out, some part of freemocap does not recognize cameras that are attached to a USB hub, at least a passive one (I will be testing a powered one soon). Given that many modern laptops have two-three built-in USB ports, I'm not sure how I can record from 6 or more cameras.

And so, I'm wondering:
- Has anyone found a way to record from 6 cameras on a laptop?
- Can freemocap work with video recorded from many different computers? These videos would have different start times - perhaps differing by as much as 1-10 seconds.

**Comment by aaroncherian** [comment, marked as answer]
Hey! Yeah so unfortunately we've had mixed success with getting USB hubs to work - I was able to get a USB hub working on my laptop, but others weren't. I'm not entirely certain if that has to do more with the specific brand of cameras or the specific USB hub. 

But, even so there seemed to be a max limit to how many cameras I could plug into the hub (above 2 cameras on a single hub it started fritzing out). So while I was able to get 4 running at once (two cameras plugged into USB ports, 2 plugged into a USB hub), I think using additional USB hubs I could've gotten more running. Generally when we've done six camera recordings, we've had cameras plugged into a desktop computer with more USB ports. 

To answer the second question - I'm not sure if you mean that you'll be running multiple webcams with FreeMoCap on multiple computers, or one webcam per computer - but the general workflow will end up being the same. FreeMoCap can work with what we call 'external recordings', which are videos from cameras that weren't all connected to a single computer running FreeMoCap, so this use case would fall broadly under that category. 

So the key here is all those videos have to be synced. When we've done this with GoPro's, we use audio syncing to sync all the videos and then cut them so they're all the same number of frames. If you can record simultaneous audio for each webcam, this as an option that would be available to you. If you're running the webcams through FreeMoCap on multiple computers however, we don't record audio so syncing becomes a little harder. You could try syncing by using a camera flash at the start of the recording and looking for the frame with the brightest light intensity for each camera.

Once you have a set of synced videos, you can take a look at this issue, I wrote out some instructions for how to run FreeMoCap with external videos: https://github.com/freemocap/freemocap/issues/255#issuecomment-1273464752

As a note, this is all in regards to the pre-alpha version. @jonmatthis might have some additional thoughts on how USB hubs work with the alpha version of the software.

**Reply by AKASGaming** [reply]
This might help with info on the USB Hubs issue
[https://docs.ipisoft.com/Cameras_and_Accessories#USB_controllers](https://docs.ipisoft.com/Cameras_and_Accessories#USB_controllers)

**Comment by mjhmilla** [comment]
Dear @aaroncherian 

Thank you very much for taking the time to write such a detailed response!

Matt
